src/platform/intel/cavs/lib/clk.c

```diff
@@ -32,7 +33,7 @@ static inline void select_cpu_clock(int freq_idx, bool release_unused)
 
 #if CONFIG_TIGERLAKE
 	if (freq_idx == CPU_HPRO_FREQ_IDX)
-		cpu_enable_core(PLATFORM_CORE_COUNT - 1);
+		pm_runtime_get(PM_RUNTIME_DSP, PLATFORM_CORE_COUNT - 1);
```

Comment:
Is it guaranteed that number of hpro clk requests is balanced by the number of de-req, so that the last core is correctly powered down when hpro is not in use?

Reply:
please check now, updated with your approach